### ISSUE TITLE
Conf  & Wandb Integration 단순화 &  RDAlpha vs. RDBeta 테스트 비교

### ISSUE THREAD

**Issue by eubinecto** [wisdomify/wisdomify#97]
## TL;DR 

... 

## Why?

- 서로 다른 버전의 RD 모델을 빠르게 비교할 수 있는 코드를 짜기 위해서. 
- 새롭게  단장한 storyteller와의 연동.
- 실험의 버전 컨트롤 더 간단하게.

## What?

새롭게 리팩토링한 storyteller에 맞추어서, conf도 재정리. 특히 데이터 부분을 정리하자.
그리고 RDBeta가 학습이 되도록 개선하기.

## To-do's
- [x] version_1은 그냥 제거하기. 현재 version_2가 지금의 version_1이 되도록 만들기
- [x] datamodule 리팩토링 -> 모든 다운로드와 로드는 여기에서 진행하기!  Tables를 여기에서 활용하면 될듯!
- [x] conf 재정비하기
- [x] ~main.train을 wandb 연결없이 할 수 있도록 만들기 [이걸 추가하면 될 것 같다](https://github.com/wandb/client/issues/764#issuecomment-730023788)~ 그냥 `upload`파라미터를 추가함. 이게 더 간편
- [x] 파이토치 random seed 고정하기
- [x] artifact를 각 RD모델을 기준으로 저장하도록 변경하기.
- [x] wisdoms의 리스트는 wandb에서 바로 가져오도록 하기.
- [x] wandb support가 지나치게 복잡하다. 정말로 필요한 기능만 남기고 리팩토링.
- [x] eg_mask를 desc_mask로 이름 변경.  desc  = either definition or an example. 이는 곧 H_eg도 H_desc로 변경한다는 것을 의미함!
- [x] wisdomifier.from_pretrained 를 하기 보단,  그냥 exp.load 를 한 것을  init인자로 받는 것이 나을 듯 하다.  
- [x] 어차피 그렇게 할 것이라면, 그냥 utils 안에 experiment & wisdomifier를 같이 넣어서 관리하는게 더 간편할 듯 싶다.
- [x] validation_end, test_end 제대로 정리하기! - pytorch-lightning의 가이드를 따라갈 것
- [x]  readme_images 없애고, 깃허브에 이미지 올리기
- [x] early stopping 구현 - val log에 대하여.. 구현하기
- [x] #98

**Comment by eubinecto** [comment]
## Artifact를 각 RD모델을 기준으로 저장하도록 변경하기
```
- RDAlpha
  - v1: latest  
  - v0
- RDBeta
  -  v2: latest
  -  v1
  -  v0
- RDBetaWithAttention
  -  v0 
- RDBetaMoreLoss
   - v0

이런식으로 정의를 하는 편이 나을 것 같다.  `conf.json`에는 항상 latest의  config를 집어넣는 것!

**Comment by eubinecto** [comment]
## Segmentation fault 오류??
```
(wisdomifyenv) (base) eubinecto@Eu-Bins-MacBook-Air wisdomify % /opt/homebrew/Caskroom/miniforge/base/lib/python3.9/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 27 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
```

음...왜 이런 오류가 나는 것일까? 아마도 메모리 부족이 원인이 아닐까 싶은데. 기존 버트 모델이 애초에  용량이 크기도 하고. 
- https://stackoverflow.com/a/65130215

배치 사이즈를 줄이면 될 것. 그런데 배치 크기가 애초에 그렇게 크지도 않은데..음...

**Comment by eubinecto** [comment]
## 데이터셋 디자인 패러다임

```
datasets.py
datamodules.py
```

를 따로 지정하는게 더 나을 것 같다.  
datasets = Wisdoms, Wisdom2Test, Wisdom2Def, Wisdom2Eg. 
torchvision 라이브러리의 디자인 패러다임을 따라하는게 좋을듯!

그리고...  앞으로 RD를 불러오고 싶을 때는, RD.from_pretrained()를 사용하기. 굳이 다운로더를 지정할필요가 없다.

**Comment by eubinecto** [comment]
## 성능향상 확인함

rd_alpha | rd_beta
--- | ---
<img width="217" alt="image" src="https://user-images.githubusercontent.com/56193069/140624896-61a13abb-fe46-4dc1-a325-444cbf1bfcc1.png"> | <img width="212" alt="image" src="https://user-images.githubusercontent.com/56193069/140624899-fc3a0a64-f80d-41fb-938f-df38dcc7e93e.png">

**Comment by eubinecto** [comment]
## Early Stopping을 구현하자!

**Comment by eubinecto** [comment]
## RDAlpha -> RDBeta 성능향상 재확인 - 이전 테스트 로직에 문제가 있었음

### trained on `wisdom2def:v0`

`rd_alpha_v0` | `rd_beta_v0`
--- | --- 
<img width="667" alt="image" src="https://user-images.githubusercontent.com/56193069/140661635-a9f82250-f742-46cf-bbc3-b8ff7d46a3fb.png"> |  <img width="656" alt="image" src="https://user-images.githubusercontent.com/56193069/140661616-094b2ba8-16cb-41af-a37e-35cdd5372c41.png">



### trained on `wisdom2def:v1`
`rd_alpha_v1` | `rd_beta_v1`
--- | --- 
<img width="650" alt="image" src="https://user-images.githubusercontent.com/56193069/140661648-4fff3834-e31d-4003-9746-dd59ce255b5d.png"> | <img width="656" alt="image" src="https://user-images.githubusercontent.com/56193069/140661543-db06afd6-1f7e-418a-bac6-5cb724425e08.png">
